### PR TITLE
Rewrite nistkatrng and calculate nistkat checksum on board

### PR DESCRIPTION
**Pull request by potsrevennil**
[//]: # (SPDX-License-Identifier: CC-BY-4.0)
[//]: # (TODO Customize PR template)

[//]: # (See https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository )

depends on #29 
- Rewrite nistkatrng to avoid license issue
- Calculate nistkat checksum on board with sha3-256 to make our life easier

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)

### CODE REVIEW

File: fips202/fips202.h [hunk 1]
```diff
@@ -55,6 +55,8 @@ void shake256_inc_finalize(shake256incctx *state);
  */
 void shake256_inc_squeeze(uint8_t *output, size_t outlen, shake256incctx *state);
 
+void sha3_256_inc_finalize(shake256incctx *state);
+
```
Comment (mkannwischer) [inline fips202/fips202.h:59]: This does not make any sense. If it's a shake256incctx, you can't create a SHA3_256 hash from that.
Also I'd prefer if you don't extend the fips202 api for the tests. 

Just use `shake256_inc_squeeze` with a 32-byte output.

**Review by mkannwischer — CHANGES_REQUESTED** [review]
Thanks. Can you please change the following?

File: fips202/fips202.h (hunk 1 above)
Comment (potsrevennil) [reply fips202/fips202.h]: I don't understand why I can replace finalize by squeeze, I thought it's more similar to absorb ?
But I moved the sha3 finalize function into the nistkat.c file avoid changing the interface of fips202, let me know if this is ok for you

Comment (mkannwischer) [reply fips202/fips202.h]: You will need both a finalize and a squeeze. For SHA3 it is only one call since the output length is fixed.

**Review by mkannwischer — APPROVED** [review]
Thanks @potsrevennil. This also looks good to me!

**Pull request merged**